ambry-tools/src/main/java/com/github/ambry/tools/perf/serverperf/ServerPerformance.java

```diff
@@ -0,0 +1,385 @@
+    networkQueue.start();
+  }
+
+  public static void main(String[] args) throws Exception {
+    VerifiableProperties verifiableProperties = ToolUtils.getVerifiableProperties(args);
+    ServerPerformance serverPerformance = new ServerPerformance(verifiableProperties);
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      try {
+        logger.info("Starting the shutdown");
+        serverPerformance.forceShutDown();
+        serverPerformance.printMetrics();
+      } catch (Exception e) {
+        logger.error("Caught error while shut down", e);
+      }
+    }));
+    serverPerformance.startGetLoadTest();
+    serverPerformance.close();
+    System.exit(0);
+  }
+
+  /**
+   * starts a thread to produce get requests
+   * starts a thread to consume responses
+   * starts a thread to start the shutdown at given time
+   * When all threads are done, {@link #shutDownLatch} is lowered
+   * so any waiting thread can continue
+   * @throws Exception
+   */
+  void startGetLoadTest() throws Exception {
+    Thread getLoadProducer = getGetLoadProducerThread();
+    Thread getLoadConsumer = getGetLoadConsumerThread();
+    Thread shutDownThread = getTimedShutDownThread();
+    shutDownThread.start();
+    getLoadProducer.start();
+    getLoadConsumer.start();
+    getLoadProducer.join();
+    getLoadConsumer.join();
+    shutDownLatch.countDown();
+  }
+
+  void printMetrics() {
+    logger.info("HTTP2 error count {}", clientMetrics.http2NetworkErrorCount.getCount());
+    logger.info("HTTP2 dropped request count {}", clientMetrics.http2RequestsToDropCount.getCount());
+    logger.info("HTTP2 send Mean rate {}", clientMetrics.http2ClientSendRate.getMeanRate());
+    logger.info("HTTP2 stream median read time {}",
+        clientMetrics.http2StreamFirstToLastFrameTime.getSnapshot().getMedian());
+    logger.info("HTTP2 stream median acquire time, {}",
+        clientMetrics.http2FirstStreamAcquireTime.getSnapshot().getMedian());
+  }
+
+  /**
+   * Waits until the {@link ServerPerformanceConfig#timeOutSeconds} to elapse
+   * or is forced out of wait and starts the shutdown
+   * @return
+   */
+  Thread getTimedShutDownThread() {
+    return new Thread(() -> {
+      try {
+        timedShutDownLatch.await(config.timeOutSeconds, TimeUnit.SECONDS);
+        logger.info("Timed shutdown triggerred");
+        shutDown();
+      } catch (Exception e) {
+        logger.error("Caught exception in shutdown thread", e);
+      }
+    });
+  }
+
+  /**
+   * Creates a thread which will keep creating requests
+   * and submit to {@link #networkQueue}
+   * @return {@link Thread}
+   */
+  Thread getGetLoadProducerThread() {
+    return new Thread(() -> {
+      boolean isShutDown = false;
+      while (!isShutDown) {
+        try {
+          isShutDown = loadProducerGETBlob();
+        } catch (Exception e) {
+          logger.error("encountered error in loadProducer", e);
+        }
+      }
```

Comment:
Let's log something out here, to indicate thread is shutdown.

Reply:
Added log.